### PR TITLE
Uplink CLI setup wizard

### PR DESCRIPTION
**Pull request by mniewrzal**
What: uplink cli wizard to define: satellite address, api key, encryption passphrase

Why: setup process should be more user-friendly and there should be a way to not define encryption passphrase with command line parameter

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: cmd/uplink/cmd/setup.go [hunk 1]
```diff
@@ -55,5 +60,60 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
-	return process.SaveConfigWithAllDefaults(cmd.Flags(), filepath.Join(setupDir, "config.yaml"), nil)
+	isInteractive = true
+	var override map[string]interface{}
+	if isInteractive {
```
Comment (coyle) [inline cmd/uplink/cmd/setup.go]: will this ever be false ?

Comment (mniewrzal) [reply cmd/uplink/cmd/setup.go]: sorry, its still wip

File: cmd/uplink/cmd/setup.go [hunk 2]
```diff
@@ -55,5 +59,53 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
-	return process.SaveConfigWithAllDefaults(cmd.Flags(), filepath.Join(setupDir, "config.yaml"), nil)
+	var override map[string]interface{}
+	if setupCfg.Interactive {
+		fmt.Print("Enter your Satellite address: ")
```
Comment (kaloyan-raev) [inline cmd/uplink/cmd/setup.go]: I think we should check if this config has been provided as a flag by the user, and not prompt for it in such case.

This way if the user passes all required config as flags, they won't need to pass `--interactive=false`.

File: cmd/uplink/cmd/root.go [hunk 3]
```diff
@@ -19,7 +19,8 @@ import (
 
 // UplinkFlags configuration flags
 type UplinkFlags struct {
-	Identity identity.Config
+	Interactive bool `help:"enable interavtive more" default:"true" setup:"true"`
```
Comment (kaloyan-raev) [inline cmd/uplink/cmd/root.go]: Typo: "interavtive"
Typo: "more" --> mode?

**Pull request merged**